### PR TITLE
Core, Parquet: Remove Parquet dictionary encoding table property

### PR DESCRIPTION
**Pull request by amogh-jahagirdar**
Follow up to https://github.com/apache/iceberg/pull/7301/files, fixes https://github.com/apache/iceberg/issues/7663

This change removes the parquet dictionary encoding for Parquet and instead uses the hadoop config ENABLE_DICTIONARY for controlling behavior in our Bloom filter test.

cc: @aokolnychyi @singhpk234 @rdblue @Fokko

### CODE REVIEW

File: parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java [hunk 1]
```diff
@@ -246,14 +245,15 @@ public <D> FileAppender<D> build() throws IOException {
...
-      boolean dictionaryEnabled = context.dictionaryEnabled();
       int dictionaryPageSize = context.dictionaryPageSize();
       String compressionLevel = context.compressionLevel();
       CompressionCodecName codec = context.codec();
       int rowGroupCheckMinRecordCount = context.rowGroupCheckMinRecordCount();
       int rowGroupCheckMaxRecordCount = context.rowGroupCheckMaxRecordCount();
       int bloomFilterMaxBytes = context.bloomFilterMaxBytes();
       Map<String, String> columnBloomFilterEnabled = context.columnBloomFilterEnabled();
+      boolean enableDictionary =
+          Boolean.valueOf(config.getOrDefault(ParquetOutputFormat.ENABLE_DICTIONARY, "true"));
```
Comment (singhpk234) [inline parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java:256]: [minor] if we going to a new line we can use : 
```java
PropertyUtil.propertyAsBoolean(config, ParquetOutputFormat.ENABLE_DICTIONARY, true);
```

File: parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java [hunk 2]
```diff
@@ -246,14 +245,15 @@ public <D> FileAppender<D> build() throws IOException {
       int rowGroupSize = context.rowGroupSize();
       int pageSize = context.pageSize();
       int pageRowLimit = context.pageRowLimit();
-      boolean dictionaryEnabled = context.dictionaryEnabled();
```
Comment (singhpk234) [inline parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java:249]: can we still keep this in context ? it's private anyways and at times can help in overriding delete and data file seperatly, thoughts ?

**Comment by singhpk234** [comment]
Thanks @amogh-jahagirdar for picking this up, apologies for my oversight in the previous pr.

File: parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java (hunk 2 above)
Comment (amogh-jahagirdar) [reply parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java:249]: Agreed, I think it makes sense, it also just follows the existing pattern a lot better and is thus more readable.

File: parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java (hunk 1 above)
Comment (amogh-jahagirdar) [reply parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java]: Updated, unfortunately still goes to a newline but it's cleaner!

**Comment by amogh-jahagirdar** [comment]
Thanks for review @singhpk234 ! I updated. No worries, was an oversight on my part during review.

**Review by singhpk234 — APPROVED** [review]
LGTM Thanks @amogh-jahagirdar !

File: docs/configuration.md [hunk 3]
```diff
@@ -54,7 +54,6 @@ Iceberg tables support table properties to configure table behavior, like the de
...
 | write.parquet.page-size-bytes                        | 1048576 (1 MB)              | Parquet page size                                                                                                                                                                                 |
 | write.parquet.page-row-limit                         | 20000                       | Parquet page row limit                                                                                                                                                                            |
- | write.parquet.dictionary.enabled                     | true                        | Enable dictionary encoding                                                                                                                                                                        |
  | write.parquet.dict-size-bytes                        | 2097152 (2 MB)              | Parquet dictionary page size                                                                                                                                                                      |
```
Comment (Fokko) [inline docs/configuration.md:57]:
```suggestion
| write.parquet.dict-size-bytes                        | 2097152 (2 MB)              | Parquet dictionary page size                                                                                                                                                                      |
```

**Review by Fokko — APPROVED** [review]

**Comment by Fokko** [comment]
Thanks for picking this up @amogh-jahagirdar and @singhpk234. I also missed it 🥴

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks for fixing this, @amogh-jahagirdar!

**Comment by aokolnychyi** [comment]
Late +1 from me too!

**Pull request merged**